nni/retiarii/execution/logical_optimizer/logical_plan.py

```diff
@@ -138,8 +141,8 @@ def _merge_graph(self, from_graph):
             new_tail = id_to_new_node[edge.tail.id]
             Edge((new_head, edge.head_slot), (new_tail, edge.tail_slot), _internal=True)._register()
 
-    def assemble(self, multi_model_placement: Dict[Model, GPUDevice]) \
-            -> Tuple[Model, Dict[Node, Union[GPUDevice, CPUDevice]]]:
+    def assemble(self, multi_model_placement: Dict[Model, Device]) \
+            -> Tuple[Model, Dict[Node, Device]]:
```

Comment:
better to add docstring for this function

Reply:
Done. I have add the docstring for `assemble` in `AbstractLogicalNode`. Also, I add comments in each type of logical node to explain its function and how should they be assembled.